### PR TITLE
adds the 'Incident' stub type

### PR DESCRIPTION
**Pull request by pbedat**
Hello!

I'm currently working on a "export to STIX2.1" feature for a client project. Their proprietary data model defines incidents, which are also defined in the new STIX2.1 spec as a "stub type".

I'd like to add this type to your lib as a starting point for property extensions.

### CODE REVIEW

**Comment by TcM1911** [comment]
Can you add an Incident in JSON format to the file `testresources/all.json`? If it's in this file, both JSONMarshal and JSONUnmarshal is tested.

**Comment by pbedat** [comment]
Thanks for coming back to me so quickly!

I've added the incident sample to `all.json`. Do you want me to add a line about this to `CONTRIBUTING.md`?

File: testresources/all.json [hunk 1]
```diff
@@ -698,6 +698,26 @@
       }
     }
   },
+  {
+    "type": "indicator",
```
Comment (TcM1911) [inline testresources/all.json:702]: Not an incident

**Review by TcM1911 — CHANGES_REQUESTED** [review]
No need to add anything to the contribution read me. I think you added the wrong JSON to the file.

File: testresources/all.json (hunk 1 above)
Comment (pbedat) [reply testresources/all.json]: Sorry, missed to add the changes in the commit 🤦‍♂️

all.json is now up to date

Comment (pbedat) [reply testresources/all.json]: Also checked the code with `go test -cover` and also added the assertions for all.json.
Coverage for this branch is now equal to master.

**Pull request merged**